tracer/src/Datadog.Trace/Sampling/SpanSamplingRule.cs

```diff
@@ -0,0 +1,159 @@
+        /// </summary>
+        /// <param name="serviceNameGlob">The glob pattern for the <see cref="Span.ServiceName"/>.</param>
+        /// <param name="operationNameGlob">The glob pattern for the <see cref="Span.OperationName"/>.</param>
+        /// <param name="samplingRate">The proportion of spans that are kept. <c>1.0</c> indicates keep all where <c>0.0</c> would be drop all.</param>
+        /// <param name="maxPerSecond">The maximum number of spans allowed to be kept per second - <see langword="null"/> indicates that there is no limit</param>
+        public SpanSamplingRule(string serviceNameGlob, string operationNameGlob, float samplingRate = 1.0f, float? maxPerSecond = null)
+        {
+            if (string.IsNullOrWhiteSpace(serviceNameGlob))
+            {
+                throw new ArgumentException("Value cannot be null or whitespace.", nameof(serviceNameGlob));
+            }
+
+            if (string.IsNullOrWhiteSpace(operationNameGlob))
+            {
+                throw new ArgumentException("Value cannot be null or whitespace.", nameof(operationNameGlob));
+            }
+
+            SamplingRate = samplingRate;
+            MaxPerSecond = maxPerSecond;
+
+            _serviceNameRegex = ConvertGlobToRegex(serviceNameGlob);
+            _operationNameRegex = ConvertGlobToRegex(operationNameGlob);
+
+            // null/absent for MaxPerSecond indicates unlimited, which is a negative value in the limiter
+            _limiter = MaxPerSecond is null ? new SpanRateLimiter(-1) : new SpanRateLimiter((int?)MaxPerSecond);
+        }
+
+        /// <inheritdoc/>
+        public float SamplingRate { get; } = 1.0f;
+
+        /// <inheritdoc/>
+        public float? MaxPerSecond { get; } = null;
+
+        /// <summary>
+        ///     Creates <see cref="SpanSamplingRule"/>s from the supplied JSON <paramref name="configuration"/>.
+        /// </summary>
+        /// <param name="configuration">The JSON-serialized configuration.</param>
+        /// <returns><see cref="IEnumerable{T}"/> of <see cref="SpanSamplingRule"/>.</returns>
+        public static IEnumerable<SpanSamplingRule> BuildFromConfigurationString(string configuration)
+        {
+            if (string.IsNullOrWhiteSpace(configuration))
+            {
+                return Enumerable.Empty<SpanSamplingRule>();
+            }
+
+            try
+            {
+                var rules = JsonConvert.DeserializeObject<List<SpanSamplingRuleConfig>>(configuration);
+                return rules?.Select(
+                                  rule => new SpanSamplingRule(
+                                      rule.ServiceNameGlob,
+                                      rule.OperationNameGlob,
+                                      rule.SampleRate,
+                                      rule.MaxPerSecond))
+                             .ToList() ?? Enumerable.Empty<SpanSamplingRule>();
```

Comment:
nit: I think we can avoid the `.ToList()` call. The `Select` call should itself return an object of type `IEnumerable<SpanSamplingRule>`

Reply:
Good catch, removed in 6f8d8cbcc9a0051dabb5ae97a8a3e47fcf62ea9d